### ISSUE TITLE
Stormland is not working

### ISSUE THREAD

**Issue by jshodanVR** [LibreVR/Revive#1379]
Stormland doesen't boot at all.
Few seconds of loading and then everything returns to Steam Home

REVIVE 1.9.1.5
STEAM

**Comment by qlyoung** [comment]
Stormland isn't out for another week, which makes it difficult to find the problem.

Would you mind posting what HMD you have?

**Comment by jshodanVR** [comment]
> Stormland isn't out for another week, which makes it difficult to find the problem.
> 
> Would you mind posting what HMD you have?

HTC VIVE, with wands controller.  nothing special. windows 10 last update, 1080 gtx, i7 8700k

**Comment by Artful72** [comment]
Is this something that the team behind Revive think they will be able to rectify fairly quickly one the game is officially released on Thursday 14th November? As they so gracefully did with Asgard's Wrath when it was released??

**Comment by Armada651** [comment]
I won't be able to this time, as I won't have access to my workstation for another month or so.

**Comment by Artful72** [comment]
Oh No that is terrible news Hope everything is ok with you Jules. May I congratulate you on how quickly you got 1.9.1 up for Asgard's Wrath will have to continue with that until you get the chance to do an updated version to allow us to play Stormland. Seasons Greetings and all that as its not far off. Regards Rob aka Artful

**Comment by jshodanVR** [comment]
Very sad news. but it's ok, we will wait dont' worry. For now i have just resumed an old CV1

**Comment by ahmanwoods** [comment]
@jshodanVR Have you tried the game with oculus hardware connected? I know that games like echo VR have issues without it.

**Comment by jshodanVR** [comment]
no i didin't onestly. ill try. keep in mind if i try to run it without any vr hardware the game run properly on monitor. but of course i can't interact.

**Comment by ahmanwoods** [comment]
@jshodanVR also is there any way you could upload a crash log for the devs?

**Comment by benz145** [comment]
I'm also having this problem on the latest stable release of Revive, attempting to play Stormland with Index. @Armada651 I've confirmed with the devs that they haven't done anything to attempt to prevent Stormland compatibility with Revive. However the game uses an in-house engine which could be related to the issue?

**Comment by Artful72** [comment]
Benz145 - Are you getting 1. The attempt to load then fails and reverts to Steam VR or as jshodan has it runs on the computer screen but not in the HMD?

**Comment by ahmanwoods** [comment]
@benz145 Could you upload a crash log here?

**Comment by hybrid11** [comment]
Damn, what a bummer.

**Comment by benz145** [comment]
> Benz145 - Are you getting 1. The attempt to load then fails and reverts to Steam VR or as jshodan has it runs on the computer screen but not in the HMD?

I've seen both. At first it was on the monitor but nothing worked. I think after I updated Revive it stopped launching all together. I confirmed that other games were working with Revive no issue.

**Comment by helifax** [comment]
Did anyone try to run the game with an older version of Revive? Ex: 1.8.1 or 1.7.3?
Perhaps it is a regression introduced in one of the later updates?

**Comment by Artful72** [comment]
Someone spoke to the Dev's (Insomniac) and they said there is no code blocking the use of ReVive with Stormland however they are using their own inhouse engine so that maybe causing the issue. Also since the update to the Nvidia drivers some have had problems running other games with ReVive that used to work fine. Its definitely weird.

**Comment by jtslade77** [comment]
NeroTheWolf requested I upload my revive Logs from attempts to launch StormLand over the past week and a half. 

Mine does the same as jshodanVR. I'm on the latest Revive. 
i7 6700K / GTX 1080. Valve Index & Rift CV1

Version: i28_20191107_1040_LIVE_FINAL_REVIEW01_Stormland_m26

ReviveInjector.txt

Launched injector with: "C:\Program Files\Revive\Revive\x64\ReviveInjector.exe" /app insomniac-games-i-28 /library "Software\insomniac-games-i-28\Game\i28.exe" -archive asset_archive/toc -level levels/stormland/stormland.level -pkg -ovr -vr_presentation -noerrormarkers -lobby -real_heroprogression { Assert.WarningFilter=review }
Creating process: S:\Rift Games\Software\insomniac-games-i-28\Game\i28.exe -archive asset_archive/toc -level levels/stormland/stormland.level -pkg -ovr -vr_presentation -noerrormarkers -lobby -real_heroprogression { Assert.WarningFilter=review } 
Injecting DLL: C:\Program Files\Revive\Revive\x64\openvr_api.dll
LoadLibrary found at address: 0xcf8beb60
Injecting DLL: C:\Program Files\Revive\Revive\x64\LibRevive64_1.dll
LoadLibrary found at address: 0xcf8beb60
Injected dlls successfully


Let me know if any other files will help. I should have the Final Release Version of StormLand today.

**Comment by LightbeamZ** [comment]
I preorderd the game and when I tried to install it today after launch, it just downloads a 8.5KB "Stormland.exe" and nothing more. If I try to launch it from the Oculus launcher it just says "device error" because no Oculus headset is connected and with Revive it just does not do anything. Looks to me like as if they have some download/installation block for people without an Oculus headset even so the Devs say that they don't.

**Comment by jtslade77** [comment]
> I preorderd the game and when I tried to install it today after launch, it just downloads a 8.5KB "Stormland.exe" and nothing more. If I try to launch it from the Oculus launcher it just says "device error" because no Oculus headset is connected and with Revive it just does not do anything. Looks to me like as if they have some download/installation block for people without an Oculus headset even so the Devs say that they don't.

The release is delayed so there is nothing to download yet or Maybe Pied Piper is using their compression algorithm on the game?

**Comment by Artful72** [comment]
The game files were not all in the launch download initially so the official launch was delayed until 9am PST/12pm ET and 5pm GMT. So around 10 minutes from me writing this. There will be an update applied to the Oculus Store. I am keeping everything crossed that the new build will just work either it will or it wont - keeping everything crossed. Artful

**Comment by LightbeamZ** [comment]
> The game files were not all in the launch download initially so the official launch was delayed until 9am PST/12pm ET and 5pm GMT. So around 10 minutes from me writing this. There will be an update applied to the Oculus Store. I am keeping everything crossed that the new build will just work either it will or it wont - keeping everything crossed. Artful

Ok. In the libary it was not showing a preorder status thats why I was confused. So release is in the next 5min so we will see by then.

**Comment by Delta2Force** [comment]
It's out btw

**Comment by LightbeamZ** [comment]
> It's out btw

Yes, but like the pre-release version not working with Revive at all. Does not start or give any signs of tring to start.

**Comment by Artful72** [comment]
I rolled back to previous versions of Revive and when I was running 1.8.0 I had a notification saying "No Rift Connected" so Im wondering if there is some kind of marker within the software/game files that reads that their is a connection and will not let you play the game without it. Just plucking at straws here. Artful

**Comment by LightbeamZ** [comment]
https://www.reddit.com/r/ValveIndex/comments/dwglcr/stormland_revive_fix/

Is this legit or is it just someone trying to infect others PCs with a virus?

**Comment by qlyoung** [comment]
That is a working patch that will be integrated into a new Revive release in the coming days. Thanks all for your input on this issue.

**Comment by CircuitLord** [comment]
It's legit, I made it :) Cross and I worked together to get it working.

**Comment by LightbeamZ** [comment]
Can Armada651 confirm this pls? Looks pretty suspicious to me?

**Comment by Artful72** [comment]
I want to thank Jules and his team for getting this up as fast as you did considering jules is on holiday you all deserve a bloody medal. Artful :)

**Comment by whyme466** [comment]
Used CircuitLord suggested .dll patch to Stormland in Revive (see https://www.youtube.com/watch?v=5rOnyApZbaU&feature=youtu.be - thanks for the quick patch!).  The game now launches via SteamVR, but some of the game's textures appear to be at the wrong depth.  Not sure if this is a Revive or game issue...

**Comment by JorgenRe1** [comment]
@whyme466 the issue you mentioned is being worked on by someone in the discord.

**Comment by ahmanwoods** [comment]
@whyme466 @JorgenRe1 I think I've narrowed the issue down to ovrLayerType_Cube not being implemented but I don't think I know enough about this stuff to implement it myself

**Comment by ahmanwoods** [comment]
I took some RenderDoc captures of the same scene on a CV1 and under Revive.

https://drive.google.com/file/d/1zUnGosqc6q28CURbmNSLg1MpdQH8HFf1/view?usp=sharing

**Comment by whyme466** [comment]
@ahmanwoods - can you describe the differences between CV1 and Revive (the download file is HUGE)?  I disassembled my Rift setup some time ago, in favor of my wireless lens-modified Vive Pro (with Index and Pimax 8K as backups).

**Comment by ahmanwoods** [comment]
@whyme466 Its the issue with the textures being at the wrong depth. I uploaded the captures in case someone wants to try to debug the issue.

**Comment by fholger** [comment]
This happens because the game engine does not correctly handle individual FOVs for each eye. I implemented a hack in Revive to work around it, see my open PR. You can find an updated Revive build here: https://github.com/fholger/Revive/releases

**Comment by whyme466** [comment]
@fholger - thanks for the fast work!  The textures are rendering correctly now, game looks great.  I am looking forward to playing this after I finish Asgard's Wrath...

**Comment by DebrisHauler** [comment]
What's the status of making stormland multiplayer work in revive? Nero on discord was showing me a hacky way to get your oculus authorization token and then send that out to oculus every 10 seconds so you remain online/authorized during play. Otherwise your multiplayer game will disconnect after a short time.

**Comment by ahmanwoods** [comment]
I have to find a good way to extract your OAuth token from the Oculus client (or patch the current running App ID). Armada indicated that they can build this into the Revive Dashboard once I have more info. 

In the mean time you can use it manually at https://github.com/ahmanwoods/Oculus-Defibrillator

**Comment by LuckMic** [comment]
> I have to find a good way to extract your OAuth token from the Oculus client (or patch the current running App ID). Armada indicated that they can build this into the Revive Dashboard once I have more info.
> 
> In the mean time you can use it manually at https://github.com/ahmanwoods/Oculus-Defibrillator

Would this also fix multiplayer problems in other games like drop dead?

**Comment by ahmanwoods** [comment]
> Would this also fix multiplayer problems in other games like drop dead?

Yeah. As long as you update the app ID in the script it fixes multiplayer for all games :)

**Comment by LuckMic** [comment]
I did not manage to get the access_token with fiddler (have you got a description how-to for dummies?), so I hope it will be part of Revive. It would be a great improvement.

**Comment by DebrisHauler** [comment]
@LuckMic I saw your original comment asked how to grab the token. Here are the steps.

Step 1. setup fiddler so it decrypts https...

fiddler tools > options > HTTPS

Make sure Decrypt HTTPS traffic is checked.

Click Actions (top right of the same HTTPS window)

Trust & install the Fiddler Root certificate

Step 2. Clear out the network traffic rows so oculus traffic will be easier to pick out.
To do this, simply stop and start Fiddler.

Step 3. start Oculus Client. You will see it generate some rows in fiddler.

Step 4. Get that acess_token!
Select any one of the rows that has graph.oculus.com in the "Host" column.
Now in the right half of fiddler, click Inspectors tab, click WebForms tab.
Left click the value that is to the right of access_token. Right click, copy to clipboard.

Step 5. Close fiddler, it can interfere with multiplayer. But leave oculus client open.

The token will be invalid the next time you stop and start oculus client.

If you wish to untrust the fiddler root cert you can read how here.
https://www.thesslstore.com/blog/how-to-remove-a-root-certificate/

**Comment by whyme466** [comment]
Finally had to chance to play more of Stormland, after finishing superb Asgard's Wrath.  Unfortunately, the Revive version of Stormland still seems to have some type of texture issues, even after fholger's patch (see above link).  Some of the plant foliage (after starting area) appear to have double/blurred texture display.

**Comment by LuckMic** [comment]
Thx for the description. I made some tests and it doesn't help for Drop Dead, which still reconnect every 3 seconds and Echo VR works perfect without it.

**Comment by ahmanwoods** [comment]
@LuckMic did you update the app ID in the script?

**Comment by fholger** [comment]
@whyme466 Hm, I haven't noticed any further issues and can't spot any obvious differences between my Index with Revive and Quest via Oculus Link. Can you perhaps point to an instance of the problem e.g. in the base camp that is easy to find?

**Comment by whyme466** [comment]
@fholger Sorry, the first place I really noticed the problem was in the immediate landing area of the Faint Signal mission, where Eco has identified a faint signal from Nix's plant.  In the immediate landing area, the small green ground covert looks out of focus, relative to all other plants.

**Comment by fholger** [comment]
Right but does it improve if you close one eye? I.e. is it really a depth issue or just blurry? Because I do see some base vegetation that's just very low-res, but I think what I'm seeing is normal and is there even on Oculus.

**Comment by RafaelSPBarbosa** [comment]
Hey, I'm not sure of this has been said before but stormlands on revive using an oculus quest with virtual desktop has some huge stuttering/jiggering on the headsets movement, it's completely unplayable.

Unfortunately from what I heard, revive's dev is on vacation right? Is there any estimate as to when this would possibly be fixed?

**Comment by whyme466** [comment]
@fholger - thanks for the suggestion and clarification.  My "problem" appears to simply be the low-res vegetation (same poor, slightly out-of-focus plants in one eye).  I hope the devs eventually replace these textures.  I have been spoiled by Asgard's Wrath, where no res textures exist...

**Comment by ahmanwoods** [comment]
@RafaelSPBarbosa Someone looked into this and its an issue on virtual desktop's side. They haven't implemented pose prediction.

**Comment by Bradgt82** [comment]
After what appears to be a revive update which is weird because I thought the guy was on vacation, I cant get Stormland to boot on Vive Cosmos.  I copied the files over from Jordans patch thing.  It still doesnt boot up and just shows a command prompt quickly and then shows it loading for a second then steam VR starts.  I reinstalled revive and pc and headset etc etc.  Nothing works for me.  I think whoever built Revive should be paid millions of dollars from Steam or all other VR makers because without the makers of revive nobody would buy any headset other then Oculus. 

Whats the point if theres nothing to play or do..  This is just stupid.  As a CV1 Oculus owner I have no idea why I decided to upgrade to new Cosmos.  I should have done more research.  Never even heard of Revive until few weeks ago.  There is no market for this crap.  It needs to be plug and play.  Anyways, yea I can barely play anything on my Cosmos through revive.  Sadly Im about to toss Cosmos in the junk and hook up my CV1 again.  I wish I'd gone with Oculus S or just not upgraded.

Revive needs to be officially supported by Steam or HTC.

**Comment by fholger** [comment]
There has been no Revive update. Try starting SteamVR first and launch the game through the Revive dashboard in SteamVR. If that doesn't work, do other games load?

**Comment by fholger** [comment]
We now have nightly builds: https://ci.appveyor.com/project/librevr/revive/build/artifacts
You can find an up-to-date installer there that will install a Revive version with working Stormland support.

**Comment by hybrid11** [comment]
Is this update to run Stormland messing with Asgard's and Vader 3.  Same version of Revive but those don't seem to run the same. Asgard's I have to launch twice before it works and Vader just doesn't detect controllers.

**Comment by fholger** [comment]
It would surprise me if the Stormland changes had any effect on other games. But did a rollback to the latest stable version make a difference for you?

**Comment by Artful72** [comment]
I seem to have lost stormland on the revive overlay any ideas as to why it would be missing when it was there ?

**Comment by theoriginalmalice** [comment]
I'd just like to say that the folks working on revive really need to market this and get paid.   I'd certainly pay a reasonable price for full time support/work from these great people.   Seriously, this is currently the closest thing that we have to "open" VR games.   Lots of talented people out there and for the rest of us we're very appreciative.  I really want to see you folks get $ without the corruption.   As for me, I'm patiently awaiting the awesomeness that is you're hard work.  I'd love to play Stormland.  

Heck, maybe I should just dive right in and capitalize.   :)

**Comment by Bradgt82** [comment]
I’m patiently waiting also for revive storm land because like a jack ass I didn’t do my research before I bought the new cosmos. I honestly have no idea why anyone would buy anything other then oculus. 

Like who would buy the new steam VR that new high end one whatever it’s called, the 1k one.  I’m not saying oculus has better tech either.. they just have all the good games apps.

**Comment by qlyoung** [comment]
Thanks everyone for your input on this issue. I will leave this issue open, pinned and locked until an official release is out for anyone looking for information. Open a separate issue if you have other bugs to report.
